### ISSUE TITLE
Courier email template by domain alias host

### ISSUE THREAD

**Issue by jdclarke5** [ory/kratos#1620]
**Is your feature request related to a problem? Please describe.**

Currently I can only `To`, `VerificationURL`, and `RecoveryURL` to define my email templates. However, I would like to change some of the email template based on the domain alias for which the verification/recovery was requested from.

**Describe the solution you'd like**

Access to a `Host` variable would be sufficient.

**Describe alternatives you've considered**

Only alternative which I can think of is to host different instances of a Kratos server with two different email templates and route traffic based on client host to each server.

**Comment by aeneasr** [comment]
We will be deprecating the domain alias feature as it makes the internal code base way too complicated and brittle.

Therefore, there probably wont be adding support for this type of configuration unfortunately.

**Comment by jdclarke5** [comment]
That is fair. In practice to make this work for us we have spun up one Kratos server for each domain, point these to the same backend, and switch internal requests based on host headers. This works without issue and scales fine given how lightweight Kratos is.

**Comment by aeneasr** [comment]
Great! That is imp the way to go :)

**Comment by MrMebelMan** [comment]
Do I understand correctly that multiple Kratos instances, configured differently, can connect to the same database without any issues?

**Comment by zepatrik** [comment]
Yes, there are some people out there who do that. It is not explicitly encouraged and might not work in a future version, but until we have e.g. #662 it is the way to go.
One important thing is to keep the clocks of the Kratos instances in sync, best done by running on the same node/VM/...

**Comment by jdclarke5** [comment]
Yes @MrMebelMan we currently find no obvious issues (using v0.6.3). Users can sign up on one domain and use these same credentials to log in on the other domain. You can be logged into both domains, or only one, at the same time, without apparent issue. We use the same secrets for both Kratos servers.

Our use case is serving basically the same application with different skins to different clients. Instead of managing multiple user spaces they all are basically signing up to a central "universe" from which they can access their bespoke skinned application at their desired domain.
